apps/nsqd/main.go

```diff
@@ -77,6 +81,17 @@ func (p *program) Start() error {
 		logFatal("failed to persist metadata - %s", err)
 	}
 
+	signalChan := make(chan os.Signal, 1)
+	go func() {
+		// range over all term signals
+		// we don't want to un-register our sigterm handler which would
+		// cause default go behavior to apply
+		for range signalChan {
+			p.nsqd.TermSignal()
+		}
+	}()
+	signal.Notify(signalChan, syscall.SIGTERM)
```

Comment:
This still feels odd to me that we have to separate out a signal to get the behavior we want. Feels like `go-svc` should be able to handle multiple signals, and "programs" can implement a callback act on a received signal and return something to `go-svc` indicating whether it should shut down?